File: project/baul-onboarding/src/main/java/fc/icd/baulonboarding/survey/model/entity/SurveyItemOption.java
package fc.icd.baulonboarding.survey.model.entity;

import fc.icd.baulonboarding.common.exception.InvalidParamException;
import fc.icd.baulonboarding.common.model.entity.AbstractEntity;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.util.StringUtils;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "survey_item_options")
public class SurveyItemOption extends AbstractEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "survey_item_id")
    private SurveyItem surveyItem;

    private String content;

    private Integer ordering;

    private boolean isDeleted;

    @Builder
    public SurveyItemOption(SurveyItem surveyItem,
                            String content,
                            Integer ordering
    ){
        if(surveyItem == null) throw new InvalidParamException("Entity SurveyItemOption.surveyItem");
        if(!StringUtils.hasText(content)) throw new InvalidParamException("Entity SurveyItemOption.content");
        if(ordering == null) throw new InvalidParamException("Entity SurveyItemOption.ordering");

        this.surveyItem = surveyItem;
        this.content = content;
        this.ordering = ordering;
    }

    public void applyChanges(String content,
                        Integer ordering,
                        Boolean isDeleted
    ){
        if(!StringUtils.hasText(content)) throw new InvalidParamException("Entity SurveyItemOption.content");
        if(ordering == null) throw new InvalidParamException("Entity SurveyItemOption.ordering");
        if(isDeleted == null) throw new InvalidParamException("Entity SurveyItemOption.isDeleted");

        this.content = content;
        this.ordering = ordering;
        this.isDeleted = isDeleted;
    }

}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/domain/SurveyItemOption.kt
package kr.innercircle.onboarding.survey.domain

import jakarta.persistence.*
import kr.innercircle.onboarding.survey.dto.request.CreateSurveyItemOptionRequest
import kr.innercircle.onboarding.survey.exception.InvalidSurveyItemTypeException

/**
 * packageName : kr.innercircle.onboarding.survey.domain
 * fileName    : SurveyItemOption
 * author      : ckr
 * date        : 25. 6. 12.
 * description :
 */

@Table(name = "survey_item_option")
@Entity
class SurveyItemOption (
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_item_id", nullable = false)
    var surveyItem: SurveyItem,

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, length = 200)
    var option: String,

    @Column(nullable = false)
    var orderNumber: Int
): BaseTimeEntity() {
    companion object {
        fun of(
            surveyItem: SurveyItem,
            createSurveyItemOptionRequest: CreateSurveyItemOptionRequest,
            orderNumber: Int
        ): SurveyItemOption {
            if(surveyItem.inputType != SurveyItemInputType.SINGLE_CHOICE && surveyItem.inputType != SurveyItemInputType.MULTIPLE_CHOICE) {
                throw InvalidSurveyItemTypeException()
            }

            return SurveyItemOption(
                surveyItem = surveyItem,
                option = createSurveyItemOptionRequest.option,
                orderNumber = orderNumber
            )
        }
    }
}